dashboard/lib/projects_list.rb

```diff
@@ -60,18 +60,99 @@ def fetch_section_projects(section)
+      published = fetch_published_project_types(PUBLISHED_PROJECT_TYPE_GROUPS.keys, limit: limit)
+      featured = fetch_featured_published_projects
+      featured_channel_ids = []
+      PUBLISHED_PROJECT_TYPE_GROUPS.keys.each do |project_type|
+        featured[project_type].each do |project|
+          featured_channel_ids << project["channel"]
+        end
+      end
+      PUBLISHED_PROJECT_TYPE_GROUPS.keys.each do |project_type|
+        published[project_type].each do |project|
+          unless featured_channel_ids.include?(project["channel"])
+            featured[project_type].push(project)
+          end
+        end
+      end
+      return featured
+    end
+
+    def fetch_featured_published_projects
+      featured_published_projects = {}
+      PUBLISHED_PROJECT_TYPE_GROUPS.keys.each do |project_type|
+        featured_published_projects[project_type] = fetch_featured_projects_by_type(project_type)
+      end
+      return featured_published_projects
+    end
+
+    def project_and_featured_project_and_user_fields
+      [
+        :storage_apps__id___id,
+        :storage_apps__storage_id___storage_id,
+        :storage_apps__value___value,
+        :storage_apps__project_type___project_type,
+        :storage_apps__published_at___published_at,
+        :featured_projects__featured_at___featured_at,
+        :featured_projects__unfeatured_at___unfeatured_at,
+        :users__name___name,
+        :users__birthday___birthday,
+        :users__properties___properties,
+      ]
+    end
+
+    def fetch_featured_projects_by_type(project_type)
+      storage_apps = "#{CDO.pegasus_db_name}__storage_apps".to_sym
+      user_storage_ids = "#{CDO.pegasus_db_name}__user_storage_ids".to_sym
+      users = "dashboard_#{CDO.rack_env}__users".to_sym
+      project_featured_project_user_combo_data = DASHBOARD_DB[:featured_projects].
+        select(*project_and_featured_project_and_user_fields).
+        join(storage_apps, id: :storage_app_id).
+        join(user_storage_ids, id: Sequel[:storage_apps][:storage_id]).
+        join(users, id: Sequel[:user_storage_ids][:user_id]).
+        where(unfeatured_at: nil).
+        where(project_type: project_type.to_s, abuse_score: 0).
+        exclude(published_at: nil).
+        order("RAND()").limit(4).all
+      extract_data_for_featured_project_cards(project_featured_project_user_combo_data)
+    end
+
+    def extract_data_for_featured_project_cards(project_featured_project_user_combo_data)
+      data_for_featured_project_cards = []
+      project_featured_project_user_combo_data.each do |project_details|
+        project_details_value = JSON.parse(project_details[:value])
+        channel = storage_encrypt_channel_id(project_details[:storage_id], project_details[:id])
+        data_for_featured_project_card = {
+          "channel" => channel,
+          "name" => project_details_value['name'],
+          "thumbnailUrl" => project_details_value['thumbnailUrl'],
```

Comment:
per earlier comment, it looks like you will need to call `StorageApps.make_thumbnail_url_cacheable` here for now.